code/modules/mob/new_player/new_player.dm

```diff
@@ -297,22 +291,67 @@
 				to_chat(src, "<span class='notice'>Vote successful.</span>")
 
 
+/mob/new_player/proc/make_me_an_observer()
+	if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
+		to_chat(src, "<span class='warning'>The game is still setting up, please try again later.</span>")
+		return FALSE
+
+	if(QDELETED(src) || !src.client)
+		ready = PLAYER_NOT_READY
+		return FALSE
+
+	var/this_is_like_playing_right = alert(src,"Are you sure you wish to observe?\nYou will have to wait at least 5 minutes before being able to respawn!","Player Setup","Yes","No")
+
+	if(QDELETED(src) || !src.client || this_is_like_playing_right != "Yes")
+		ready = PLAYER_NOT_READY
+		src << browse(null, "window=playersetup") //closes the player setup window
+		new_player_panel()
+		return FALSE
+
+	var/mob/dead/observer/observer = new()
+	spawning = TRUE
+
+	observer.started_as_observer = TRUE
+	close_spawn_windows()
+	var/obj/effect/landmark/observer_start/O = locate(/obj/effect/landmark/observer_start) in GLOB.landmarks_list
+	to_chat(src, "<span class='notice'>Now teleporting.</span>")
+	if (O)
+		observer.forceMove(O.loc)
+	else
+		to_chat(src, "<span class='notice'>Teleporting failed. Ahelp an admin please</span>")
+		stack_trace("There's no freaking observer landmark available on this map or you're making observers before the map is initialised")
+	observer.key = key
+	observer.client = client
+	//observer.set_ghost_appearance()
+	if(observer.client && observer.client.prefs)
```

Comment:
```suggestion
	if(observer.client?.prefs)
```

Reply:
preferably making it more readable

Reply:
It's the standard here and hopefully soonTM should also be in the contributor guidelines, since we already demand from people to do it this way, if it's possible in the circumstance.